File: frontend/src/main/typescript/backend/dto/response/queryConsoleResponse.ts
import { UpdateErrorReport, VocaDbTag } from "@/backend/dto/lowerLevelStruct";
import { ArtistType, SongType } from "@/backend/dto/enumeration";

export interface QueryConsoleResponse {
  items: (QueryConsoleArtistItem | QueryConsoleSongItem)[];
  totalCount: number;
  tagPool: VocaDbTag[];
}

export interface QueryConsoleArtistItem {
  id: number;
  name: string;
  tags: VocaDbTag[];
  type: ArtistType;
  tagsToRemove: VocaDbTag[];
  errorReport: UpdateErrorReport | null;
}

export interface QueryConsoleSongItem {
  id: number;
  name: string;
  tags: VocaDbTag[];
  type: string;
  artistString: string;
  tagsToRemove: VocaDbTag[];
  errorReport: UpdateErrorReport | null;
}

File: backend/src/main/kotlin/niconicotagger/dto/api/response/QueryConsoleResponse.kt
package niconicotagger.dto.api.response

import niconicotagger.dto.api.misc.QueryConsoleData
import niconicotagger.dto.inner.vocadb.VocaDbTag

data class QueryConsoleResponse<T : QueryConsoleData>(
    override val items: List<T>,
    val tagPool: List<VocaDbTag>,
    override val totalCount: Long
) : ApiResponse<T>
